### PR TITLE
Add Multiple Media Form Field

### PR DESCRIPTION
**Pull request by AlexGStapleton**
Resolve https://github.com/siteorigin/so-widgets-bundle/issues/877

Screenshot:

### CODE REVIEW

**Comment by AlexGStapleton** [comment]
I've created a quick test wide to test this form field. You can download it by [clicking here](https://github.com/siteorigin/so-widgets-bundle/files/5942309/siteorigin-multiple-media-widget.zip). Once installed navigate to **Plugins > SiteOrigin Widgets** and activate the **SiteOrigin Multiple Media Test** widget. The widget will output an array on the frontend.

**Comment by AlexGStapleton** [comment]
The last commit changes the Add Media button to use the WP .button styling. This styling respects the users selected Admin Color Scheme.

Screenshot_2021-02-10 Edit Page ‹ SiteOrigin — WordPress(2)

**Comment by AlexGStapleton** [comment]
I've logged an issue about introducing repeater support in a future update:

https://github.com/siteorigin/so-widgets-bundle/issues/1250

File: base/inc/fields/js/multiple-media-field.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+			var $currentItem = $( this ).parent();
+
+			selectedMedia.splice( selectedMedia.indexOf( $currentItem.data( 'id' ) ) );
+			$data.val( selectedMedia.join( ',' ) );
+
+			$currentItem.remove();
+		} );
+
+		$field.data( 'initialized', true );
+	});
```
Comment (Misplon) [inline base/inc/fields/js/multiple-media-field.js:127]: Spacing for `});`.

File: base/inc/fields/js/multiple-media-field.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+			// Create the media frame.
+			frame = wp.media( {
+				title: $$.data( 'choose' ),
+				library: {
+					type: $$.data( 'library' ).split( ',' ).map( function( v ){ return v.trim(); })
+				},
+				multiple: true,
+				button: {
+					text: $$.data('update'),
```
Comment (Misplon) [inline base/inc/fields/js/multiple-media-field.js:38]: Spacing for `('update'),`.

File: base/inc/fields/js/multiple-media-field.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+			if ( frame ) {
+				frame.open();
+				return false;
+			}
+
+			// Create the media frame.
+			frame = wp.media( {
+				title: $$.data( 'choose' ),
+				library: {
+					type: $$.data( 'library' ).split( ',' ).map( function( v ){ return v.trim(); })
```
Comment (Misplon) [inline base/inc/fields/js/multiple-media-field.js:34]: Spacing required for `function( v ){ return v.trim(); }`.

File: base/inc/fields/js/multiple-media-field.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			selectedMedia = $data.val().split( ',' );
+
+		if ( $field.data( 'initialized' ) ) {
+			return;
+		}
+
+		// Handle the media uploader
+		$field.find( '.button' ).on( 'click', function( e ) {
+			e.preventDefault();
+			if( typeof wp.media === 'undefined' ) {
```
Comment (Misplon) [inline base/inc/fields/js/multiple-media-field.js:17]: A space is needed after `if`.

File: base/inc/fields/multiple-media.class.php [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	}
+
+	protected function render_field( $attachments, $instance ) {
+		if ( version_compare( get_bloginfo('version'), '3.5', '<' ) ){
+			printf( __( 'You need to <a href="%s">upgrade</a> to WordPress 3.5 to use media fields', 'so-widgets-bundle'), admin_url('update-core.php' ) );
+			return;
+		}
+
+		// If library is set to all, convert it to a wildcard as all isn't valid
+		if( $this->library == 'all' ){
```
Comment (Misplon) [inline base/inc/fields/multiple-media.class.php:49]: Spacing here.

File: base/inc/fields/multiple-media.class.php [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+						</div>
+					<?php
+					}
+				}
+				?>
+			</div>
+			
+			<div class="multiple-media-field-template" style="display:none">
+				<div class="multiple-media-field-item">
+					<img class="thumbnail"/>
```
Comment (Misplon) [inline base/inc/fields/multiple-media.class.php:91]: A space before `/>`.

File: base/inc/fields/multiple-media.class.php [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+					<a href="#" class="media-remove-button"><?php esc_html_e( 'Remove', 'so-widgets-bundle' ) ?></a>
```
Comment (Misplon) [inline base/inc/fields/multiple-media.class.php:92]: Should we close `esc_html_e( 'Remove', 'so-widgets-bundle' )`? `;`

File: base/inc/fields/multiple-media.class.php [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+		$value = explode( ',', $value );
+		$media = array();
+
+		foreach ( $value as $item ) {
+			$media[] = (int) $item;
+		}
+		return $media;
+	}
+
+	function enqueue_scripts(){
```
Comment (Misplon) [inline base/inc/fields/multiple-media.class.php:118]: Spacing for `(){`.

File: base/inc/fields/multiple-media.class.php (hunk 7 above)
Comment (AlexGStapleton) [reply base/inc/fields/multiple-media.class.php:92]: May as well.

**Review by Misplon — APPROVED** [review]

**Review by gregpriday — APPROVED** [review]
Can't see any issues with this code. As usual, as long as it's been well tested, we're all good to go here.

**Pull request merged**